packages/experimental/src/components/APIKeyModal/APIKeyModal.js

```diff
@@ -0,0 +1,236 @@
+          className={`${expPrefix}--apikey-modal-loader`}
+        />
+      )}
+      {apiKeyLoaded && (
+        <div className={`${expPrefix}--apikey-modal-messaging`}>
+          <InformationFilled16 />
+          {downloadable ? (
+            <APIKeyDownloader
+              apiKey={apiKey}
+              bodyText={downloadBodyText}
+              fileName={downloadableFileName}
+              linkText={downloadLinkText}
+            />
+          ) : (
+            <div className={`${expPrefix}--apikey-modal-messaging-text`}>
+              {successBody}
+            </div>
+          )}
+        </div>
+      )}
+    </Modal>
+  );
+};
+
+APIKeyModal.propTypes = {
+  /**
+   * the api key the user recieves
+   */
+  apiKey: PropTypes.string,
+  /**
+   * id for the api key input field
+   */
+  apiKeyInputId: PropTypes.string,
+  /**
+   * label for the api key input field
+   */
+  apiKeyLabel: PropTypes.string,
+  /**
+   * text for the copy button
+   */
+  copyButtonText: PropTypes.string,
+  /**
+   * button text for the create key button
+   */
+  createButtonText: PropTypes.string,
+  /**
+   * modal header for the create a key screen
+   */
+  createHeader: PropTypes.string,
+  /**
+   * text fot the download message
+   */
+  downloadBodyText: PropTypes.string,
+  /**
+   * text for the download link
+   */
+  downloadLinkText: PropTypes.string,
+  /**
+   * specifies if the api key is downloadable or not
+   */
+  downloadable: PropTypes.bool,
+  /**
+   * specifies the filename for downloadable api keys
+   */
+  downloadableFileName: PropTypes.string,
+  /**
+   * specifies if the api key creation is loading
+   */
+  loading: PropTypes.bool,
+  /**
+   * loading message for when the api key is loading
+   */
+  loadingMessage: PropTypes.string,
+  /**
+   * content for modal body
+   */
+  modalBody: PropTypes.string,
+  /**
+   * helper text for name input
+   */
+  nameHelperText: PropTypes.string,
+  /**
+   * id for name input
+   */
+  nameInputId: PropTypes.string,
+  /**
+   * label for name input
+   */
+  nameLabel: PropTypes.string,
+  /**
+   * placeholder for name input
+   */
+  namePlaceholder: PropTypes.string,
+  /**
+   * specifices if a name is required or not
+   */
+  nameRequired: PropTypes.bool,
+  /**
+   * function to close the modal
+   */
+  onRequestClose: PropTypes.func,
+  /**
+   * function that is called to create the api key
+   */
+  onRequestSubmit: PropTypes.func,
+  /**
+   * specifies if the modal is open or not
+   */
+  open: PropTypes.bool,
+  /**
+   * text for the close button
+   */
+  secondaryButtonText: PropTypes.string,
+  /**
+   * content for when an api key is created successfully
+   */
+  successBody: PropTypes.node,
+  /**
+   * modal header for a successful api key creation
+   */
+  successHeader: PropTypes.string,
+  /**
+   * specifies if the api key input should use the password type toggle
+   */
+  visibilityToggler: PropTypes.bool,
```

Comment:
A poor variable name implies a method to be called.
```suggestion
 aipKeyVisibility: PropTypes.bool,
```